app/src/routes/api/v1/query.router.js

```diff
@@ -129,15 +128,29 @@ class QueryRouter {
 
 }
 
-const deserializeDataset = async (ctx, next) => {
-    logger.debug('Body', ctx.request.body);
-    if (ctx.request.body.dataset && ctx.request.body.dataset.data) {
-        ctx.request.body.dataset = await new JSONAPIDeserializer({
-            keyForAttribute: 'camelCase'
-        }).deserialize(ctx.request.body.dataset);
-    } else if (ctx.request.body.dataset && ctx.request.body.dataset.table_name) {
-        ctx.request.body.dataset.tableName = ctx.request.body.dataset.table_name;
+const getDatasetById = async (ctx, next) => {
```

Comment:
Any particular reason this method needs to be duplicated here (additionally to the dataset router)? 
I'd prefer this method to be imported from a `middlewares` directory that could share the code for both routers, so that we avoid unnecessary code duplication :)